### DISCUSSION TITLE
consult-grep to occur-mode

### DISCUSSION THREAD

**Discussion by rakanalh** [minad/consult#1093] [Q&A] [answered]
Hello,

Would it be possible to support exporting consult-grep results to occur mode?

The use case consists of searching through the buffer / project, using occur mode to delete specific results and then:
1. either converting the buffer into a wgrep buffer where search and replace can be performed.
2. or just navigating the results knowing that i don't have to scroll through the ones i don't need to navigate into.

Thanks

**Comment by oantolin** [comment, marked as answer]
You an currently export consult-grep results to a grep-mode buffer. It sounds to me like that covers both your use cases well, why do you want occur specifically?

**Comment by rakanalh** [comment]
grep-mode buffer is ready-only and does not offer a way to edit the list by removing search candidates. A use case for this would be something as follows:
- i would like to search for `pub fn` and convert them to `pub async fn` (dumb example but bare with me). So we perform the search and assuming we get the occur buffer.
- Remove all candidates which are tests functions
- Convert the resulting list in occur into wgrep (that was possible using ivy-counsel)
- Search / replace .. done.

**Comment by rakanalh** [comment]
I just realized that `keep-lines` and `flush-lines` can be used to perform step 2 of this. The rest is already implemented i guess.
